streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -426,6 +472,6 @@ private String repartitionIfRequired(final String queryableStoreName) {
         if (!repartitionRequired) {
             return this.name;
         }
-        return KStreamImpl.createReparitionedSource(this, keySerde, valSerde, queryableStoreName);
+        return KStreamImpl.createReparitionedSource(this.builder, keySerde, valSerde, queryableStoreName, name);
```

Comment:
So we need `this` here? (maybe also fix for `this.name` above?)

Reply:
no, we don't need it.